src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -447,7 +452,11 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
 
                     case 'h':
                         tokenLen = ParseRepeatPattern(format, i, ch);
-                        hour12 = dateTime.Hour % 12;
+                        hour12 = dateTime.Hour;
+                        if (hour12 >= 12)
+                        {
+                            hour12 -= 12;
+                        }
                         if (hour12 == 0)
                         {
                             hour12 = 12;
```

Comment:
To avoid 12 -> 0 -> 12
```cs
if (hour12 > 12)
{
    hour12 -= 12;
}
if (hour12 == 0)
{
    hour12 = 12;
}
```

or maybe even

```cs
if (hour12 > 12)
{
    hour12 -= 12;
}
else if (hour12 == 0)
{
    hour12 = 12;
}
```